hooks/expand_show_tree.py

```diff
@@ -0,0 +1,125 @@
+"""
+This hook is called when an engine, app or framework calls 
+
+> self.expand_show_tree(path)
+
+Typically apps, engines and frameworks call this method
+when they want to map the given path on a show tree.
+The default implementation expands this path on the basis of the context of ``bundle_obj``
+And would resolve to either shot/sequence/show depending on which folders exist,
+"$DD_SHOWS_ROOT/$DD_SHOW/$DD_SEQ/$DD_SHOT",
+"$DD_SHOWS_ROOT/$DD_SHOW/$DD_SEQ",
+"$DD_SHOWS_ROOT/$DD_SHOW",
+This would typically not needed to be customized.
+
+In case customization is required, the hook is passed the app/engine/framework
+that issued the original request - this gives access to configuration,
+app methods, environment etc. and should allow for some sophisticated
+introspection inside the hook.
+"""
+
+import os
+
+import sgtk
+from sgtk import dd_xplatform_utils
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class ExpandShowTree(HookBaseClass):
+    
+    def execute(self, path, additional_subdirectories, bundle_obj, **kwargs):
+        """
+        Handle expansion of the given path on show tree, depending on context issued from an app, framework or engine.
+        
+        :param path: path to create
+        :param additional_subdirectories: additional subdirectories to be included in the show hierarchy
+        :param bundle_obj: object requesting the creation. This is a legacy
+                              parameter and we recommend using self.parent instead.
+
+        :return: A list of paths in the order of Test_User:Shot:Seq:Project,
+        code that calls it can decide what to do with that paths.
+        """
+        # this is (:) separated list of paths
+        show_paths = self.expand_tree(bundle_obj, additional_subdirectories, path)
+        paths_list = show_paths.split(":")
+
+        # let's return the path list and let the code that calls it decide what to do
+        return paths_list
+
+        # we need to return the first path since that is the override we have configured
+        # Seq:Shot:Project
+        # if paths_list:
+        #     return paths_list[0]
+
+        # none of the paths exist! the hook was configured without actually populating the show with the hook.
+        # return ''
+
+    @staticmethod
+    def expand_tree(bundle_obj, additional_subdirectories, *paths):
```

Comment:
This method should return a list of paths, rather than a colon delimited path, to match the `dd.runtime.expandTree()` functionality.

Reply:
`dd.xplatform.xjoin` returns the colon separated paths, since those are search paths. The hook returns a list of paths this is just a helper method.

Reply:
Yes, but the helper method, which could potentially get invoked independently of the hook path resolution, should operate consistently with its namesake. This will also help with the transition later on to the DD pipeline.

Reply:
Nevermind, just noticed there are two versions of expandTree, `dd.runtime.api.Environment.current().expandTree()` and `dd.runtime.Context.expandTree()`. Looks like this is mirroring the former, which does indeed return a colon-delimited string...

Reply:
Yes, that's the one I was following `dd.runtime.api.Environment.current().expandTree()` so I will retain the current behavior of colon separated search paths.